src/lists.rs

```diff
@@ -313,26 +311,33 @@ where
             DefinitiveListTactic::Horizontal if !first => {
                 result.push(' ');
             }
-            DefinitiveListTactic::FormatCall if !first => {
-                result.push('\n');
-                result.push_str(indent_str);
-                tactic = DefinitiveListTactic::Horizontal;
-            }
-            DefinitiveListTactic::WriteCall => {
-                let second = i == 1;
-                let third = i == 2;
-
-                if first {
+            DefinitiveListTactic::SpecialMacro(
+                one_line_before,
+                one_line_after,
+                num_args_before,
+            ) => {
+                if i == 0 {
                     // Nothing
-                } else if second {
+                } else if i < num_args_before {
+                    if one_line_before {
+                        result.push(' ');
+                    } else {
+                        result.push('\n');
+                        result.push_str(indent_str);
+                    }
+                } else if i == num_args_before {
```

Comment:
Would you please combine `i == num_args_before` and `i == num_args_before + 1` into a single `else if` block?